src/wp-includes/class-wp-scripts.php

```diff
@@ -281,6 +281,24 @@ public function do_item( $handle, $group = false ) {
 			$intended_strategy = '';
 		}
 
+		// If the script is in the header group, current output is the header, the intended strategy is `defer`, the
+		// actual strategy is not defer and all dependent scripts are in the footer - move this script to the footer.
+		if ( 'defer' === $intended_strategy && 'defer' !== $strategy && 0 === $this->groups[ $handle ] && 0 === $group ) {
+			// Check that all depedencies are in the footer.
+			$all_deps_in_footer = true;
+			foreach ( $this->get_dependents( $handle ) as $dep ) {
+				if ( isset( $this->groups[ $dep ] ) && 1 !== $this->groups[ $dep ] ) {
+					$all_deps_in_footer = false;
+					break;
+				}
+			}
+			if ( $all_deps_in_footer ) {
+				$this->in_footer[] = $handle;
+				$this->groups[ $handle ] = 1;
```

Comment:
There is similar code above which prevents printing an `$in_footer` script in the `head`:

https://github.com/WordPress/wordpress-develop/blob/a4f6bccb134b251483777a47259f0593bec00237/src/wp-includes/class-wp-scripts.php#L252-L255

Should this logic be incorporated into that existing location?

Reply:
IMO ok to keep here since there is a lot of other logic in between. If we were to combine this, I think it would lead to more line changes than we _need_ to touch here.

Reply:
I did alter to the code to match that instance, namely: don't set group and return false.